track_info.c

```diff
@@ -52,16 +48,18 @@ struct track_info *track_info_new(const char *filename)
 	uint64_t uid = atomic_fetch_add_explicit(&cur_uid, 1, memory_order_relaxed);
 	BUG_ON(uid == 0);
 
-	struct track_info *ti;
-	ti = xnew(struct track_info, 1);
-	ti->uid = uid;
+	struct track_info_priv *priv = xnew(struct track_info_priv, 1);
+	atomic_store_explicit(&priv->ref_count, 1, memory_order_relaxed);
```

Comment:
I don't think atomics should be used with plain operations it's unnecessary magic that's also a bit misleading. E.g. in the functions above you get a synchronization barrier, but consider:

```c
void h(struct X *x)
{
	*x = (struct X) { 0 };
}
```

```asm
h:
	movl	$0, (%rdi)
	retq
```

No barrier if the plain assignment is made as part of a compound assignment.